internal/controller/dnsrecord_controller_test.go

```diff
@@ -99,15 +99,15 @@ var _ = Describe("DNSRecordReconciler", func() {
 			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
 		}
 		if dnsRecord2 != nil {
-			err := k8sClient.Delete(ctx, dnsRecord)
+			err := k8sClient.Delete(ctx, dnsRecord2)
 			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
 		}
 		if managedZone != nil {
-			err := k8sClient.Delete(ctx, managedZone)
+			err := k8sClient.Delete(ctx, managedZone, client.PropagationPolicy(metav1.DeletePropagationForeground))
 			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
 		}
 		if brokenZone != nil {
-			err := k8sClient.Delete(ctx, brokenZone)
+			err := k8sClient.Delete(ctx, brokenZone, client.PropagationPolicy(metav1.DeletePropagationForeground))
```

Comment:
Not sure these make any difference unless you set the owner ref up with block owner deletion https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#ownership-and-finalizers